fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/rules/InComparisonToExplodeRule.java

```diff
@@ -0,0 +1,140 @@
+import com.apple.foundationdb.record.Bindings;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.query.expressions.Comparisons;
+import com.apple.foundationdb.record.query.plan.temp.CorrelationIdentifier;
+import com.apple.foundationdb.record.query.plan.temp.GroupExpressionRef;
+import com.apple.foundationdb.record.query.plan.temp.PlannerRule;
+import com.apple.foundationdb.record.query.plan.temp.PlannerRuleCall;
+import com.apple.foundationdb.record.query.plan.temp.Quantifier;
+import com.apple.foundationdb.record.query.plan.temp.expressions.ExplodeExpression;
+import com.apple.foundationdb.record.query.plan.temp.expressions.SelectExpression;
+import com.apple.foundationdb.record.query.plan.temp.matchers.BindingMatcher;
+import com.apple.foundationdb.record.query.plan.temp.matchers.PlannerBindings;
+import com.apple.foundationdb.record.query.plan.temp.matchers.ValueMatchers;
+import com.apple.foundationdb.record.query.predicates.AndPredicate;
+import com.apple.foundationdb.record.query.predicates.LiteralValue;
+import com.apple.foundationdb.record.query.predicates.QuantifiedColumnValue;
+import com.apple.foundationdb.record.query.predicates.QueryPredicate;
+import com.apple.foundationdb.record.query.predicates.ValuePredicate;
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+
+import static com.apple.foundationdb.record.query.plan.temp.matchers.MultiMatcher.all;
+import static com.apple.foundationdb.record.query.plan.temp.matchers.MultiMatcher.some;
+import static com.apple.foundationdb.record.query.plan.temp.matchers.QuantifierMatchers.forEachQuantifier;
+import static com.apple.foundationdb.record.query.plan.temp.matchers.QueryPredicateMatchers.anyComparisonOfType;
+import static com.apple.foundationdb.record.query.plan.temp.matchers.QueryPredicateMatchers.valuePredicate;
+import static com.apple.foundationdb.record.query.plan.temp.matchers.RelationalExpressionMatchers.selectExpression;
+
+/**
+ * TODO
+ *
+ * A simple rule that performs some basic Boolean normalization by flattening a nested {@link AndPredicate} into a single,
+ * wider AND. This rule only attempts to remove a single {@code AndComponent}; it may be repeated if necessary.
+ * For example, it would transform:
+ * <code>
+ *     Query.and(
+ *         Query.and(Query.field("a").equals("foo"), Query.field("b").equals("bar")),
+ *         Query.field(c").equals("baz"),
+ *         Query.and(Query.field("d").equals("food"), Query.field("e").equals("bare"))
+ * </code>
+ * to
+ * <code>
+ *     Query.and(
+ *         Query.field("a").equals("foo"),
+ *         Query.field("b").equals("bar"),
+ *         Query.field("c").equals("baz")),
+ *         Query.and(Query.field("d").equals("food"), Query.field("e").equals("bare"))
+ * </code>
```

Comment:
I am guessing this is here as a template for how to structure the actual documentation and sample code. Because this isn't really what this class does.

Reply:
Sorry, yes! Updating.